mlir/lib/QEC/Transforms/CountPPMSpecs.cpp

```diff
@@ -0,0 +1,109 @@
+// Copyright 2025 Xanadu Quantum Technologies Inc.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#define DEBUG_TYPE "ppm_specs"
+
+#include <algorithm>
+#include <string>
+
+#include <nlohmann/json.hpp>
+
+#include "mlir/IR/PatternMatch.h"
+#include "mlir/Pass/Pass.h"
+
+#include "QEC/IR/QECDialect.h"
+#include "Quantum/IR/QuantumOps.h"
+
+using namespace llvm;
+using namespace mlir;
+using namespace catalyst;
+using namespace catalyst::qec;
+using json = nlohmann::json;
+
+namespace catalyst {
+namespace qec {
+
+#define GEN_PASS_DEF_COUNTPPMSPECSPASS
+#define GEN_PASS_DECL_COUNTPPMSPECSPASS
+#include "QEC/Transforms/Passes.h.inc"
+
+struct CountPPMSpecsPass : public impl::CountPPMSpecsPassBase<CountPPMSpecsPass> {
+    using CountPPMSpecsPassBase::CountPPMSpecsPassBase;
+
+    void print_specs()
+    {
+        llvm::BumpPtrAllocator string_allocator;
+        llvm::DenseMap<StringRef, int> PPM_Specs;
+        PPM_Specs["num_logical_qubits"] = 0;
+        PPM_Specs["num_of_ppm"] = 0;
+
+        // Walk over all operations in the IR (could be ModuleOp or FuncOp)
+        getOperation()->walk([&](Operation *op) {
+            // Skip top-level container ops if desired
+            if (isa<ModuleOp>(op))
+                return;
+
+            StringRef gate_name = op->getName().getStringRef();
+
+            if (gate_name == "quantum.alloc") {
```

Comment:
```suggestion
            if (isa<quantum::AllocOp>(op)) {
```

Usually it's better to use `isa` for these kind of checks since op name might change. 
Same for the others down there.